### DISCUSSION TITLE
Hyperhdr forgotten password

### DISCUSSION THREAD

**Discussion by Marz1974** [awawa-dev/HyperHDR#641] [Q&A] [answered]
Hi guys.

This is my 1st build.got leds, pi 3b etc .hyper hdr installed. It used to work but doesn't now. I've forgotten hyper hdr password...... any help appreciated

**Comment by awawa-dev** [comment]
Hi
Did you restart rpi and you still cannot log in? Then ssh to Rpi. If you use standard `pi` user:
```
sudo systemctl stop hyperhdr@pi.service 
/usr/bin/hyperhdr --resetPassword
sudo systemctl start hyperhdr@pi.service
```

**Reply by Marz1974** [reply]
Yeah I restarted it. No joy will try this later.
Thanks

**Reply by awawa-dev** [reply]
OK, just make sure HyperHDR is running as `pi` user before you try that (running HyperHDR service e.g. as `root` requires a bit different commands)

**Reply by Marz1974** [reply]
Hi tried it just there now
![16952014323349144778192179024102](https://github.com/awawa-dev/HyperHDR/assets/44277732/6453e84e-169a-4830-9005-d86893bf933b)

**Reply by Marz1974** [reply]
But when I open hyperhdr in Web browser I get this
![16952015255835226292807586855241](https://github.com/awawa-dev/HyperHDR/assets/44277732/7edcaa62-d3c9-4ab3-b7a4-12c75b1ba663)

**Reply by awawa-dev** [reply]
Tell me..are you running HyperHDR as a root? You should run each of command separatly, one by one as I wrote them, not all of them at once.

**Reply by Marz1974** [reply]
HDR loads up as soon as pi switched on

**Reply by awawa-dev** [reply]
What's your HyperHDR LED driver you are using?

**Reply by Marz1974** [reply]
not sure...sorry
ive done the lines seperatly
then go into hdr web browser bit and still asking for password

**Reply by awawa-dev** [reply]
What type of LEDs are you using? Where is the LED strip connected? Directly to Rpi?

**Reply by Marz1974** [reply]
rextin ws2812b LED
strip is connected to power supply is shares it with the pi
https://youtu.be/J26oYlKyq7Q?si=o0qdn4tfio3tu3LO
this is the video i followed

**Reply by awawa-dev** [reply]
well, so you are running HyperHDR as a root after all. So the commands should be different as I warned at the beginning..
1 restart RPi!!!
2 `sudo systemctl stop hyperhdr@root.service` 
Important because I don't know how your service was created when you've followed that guide: check now, you should not be able to connect web browser to HyperHDR now. If HyperHDR has stopped then it's OK.
3 `sudo /usr/bin/hyperhdr --resetPassword`
take a screenshot of the result
4 `sudo systemctl start hyperhdr@root.service` HyperHDR should be running again.

**Reply by Marz1974** [reply]
???
![20230920_143308](https://github.com/awawa-dev/HyperHDR/assets/44277732/229cb9c8-2c38-40d4-81bf-0b5db07eceeb)

**Reply by Marz1974** [reply]
![20230920_143502](https://github.com/awawa-dev/HyperHDR/assets/44277732/93458c17-ad48-47e2-9fca-17fd7bb34aa3)

**Reply by Marz1974** [reply]
![20230920_143726](https://github.com/awawa-dev/HyperHDR/assets/44277732/c2edd32e-b5ab-45f3-adaa-bf0d5a9f8053)

**Reply by Marz1974** [reply]
![20230920_143843](https://github.com/awawa-dev/HyperHDR/assets/44277732/07fe566e-597a-4728-8c94-d16bac0261da)

**Reply by Marz1974** [reply]
think i might wipe sd card and reinstall

**Reply by awawa-dev** [reply, marked as answer]
If you have follow that tutorial on youtube then unfortunately you have messed up with the services. I don't have control on how the people make them. Before reinstalling everything from scratch try:
1 sudo reboot 
2 after boot
```
sudo systemctl stop hyperhdr@pi.service 
sudo systemctl disable hyperhdr@pi.service 
sudo systemctl stop hyperhdr@root.service
sudo /usr/bin/hyperhdr --resetPassword
```

3 run `sudo /usr/bin/hyperhdr` check if you can connect to HyperHDR
4 If it works control-c in the terminal to stop it. Next:
`sudo systemctl enable hyperhdr@root.service`
`sudo systemctl start hyperhdr@root.service`

**Reply by Marz1974** [reply]
All went well....  until
![20230920_152329](https://github.com/awawa-dev/HyperHDR/assets/44277732/739943ba-64aa-4fb8-b06b-21358de2979f)

**Reply by awawa-dev** [reply]
If you are OK with installing from scratch then follow this tutorial: https://github.com/awawa-dev/HyperHDR/wiki/Installation#raspberrypi-os-sd-card-image-installation After you install it, you also should run HyperHDR as a root for this LED strip. There is a link how to switch HyperHDR to root user https://github.com/awawa-dev/HyperHDR/wiki/FAQ (otherwise mmap will fail) and how to configure your LED strip if you insist on direct driving ws2812 from Rpi. Still, not recommended or officially supported solution because there are better alternatives.

**Reply by Marz1974** [reply]
i tried it again   before i followed point 4 i was able to get into hyperhdr!!!  but couldnt CHANGE password......thankyou ever soo much for your help... great help....have you any better alternatives?
ive enjoed this problem solving excercise

**Reply by awawa-dev** [reply]
You can try to clean the cookies in the browser for your HyperHDR site address. Maybe there is something that interferes correct work? I dont have other ideas: reinstallation or deleting you HyperHDR configuration database when the service is stopped.
